### ISSUE TITLE
SUSEPrime say it is NVIDIA, but system info say AMD Renoir

### ISSUE THREAD

**Issue by kevinsmia1939** [openSUSE/SUSEPrime#78]
Hi,

I use openSUSE Tumbleweed latest snapshot.
Lenovo Legion 5
nVidia TU117M [GeForce GTX 1650 Ti Mobile]

From SUSEPrime, NVIDIA driver is in use.
But nvidia-settngs cannot be open because it say nvidia driver not loaded.
```
andy@localhost:~> sudo prime-select nvidia
[sudo] password for root: 
nvidia catched
nvidia driver already in use!
andy@localhost:~> /usr/sbin/prime-select get-current
Driver configured: nvidia
bbswitch module not found. NVIDIA card will not be powered off
andy@localhost:~> nvidia-settings
ERROR: NVIDIA driver is not loaded
ERROR: Unable to load info from any available system
```

But looking at KDE Info center, it say AMD Renoir (integrated graphic), no nvidia.
```
Operating System: openSUSE Tumbleweed 20220312
KDE Plasma Version: 5.24.3
KDE Frameworks Version: 5.91.0
Qt Version: 5.15.2
Kernel Version: 5.16.11-1-default (64-bit)
Graphics Platform: X11
Processors: 12 × AMD Ryzen 5 4600H with Radeon Graphics
Memory: 7.1 GiB of RAM
Graphics Processor: AMD RENOIR
```

NVIDIA G05 driver is installed
Version: 470.103.01-52.1
Repo: obs://build.suse.de/Proprietary:X11:Drivers

```
andy@localhost:~> sudo zypper se G05
Loading repository data...
Reading installed packages...

S  | Name                      | Summary                                                               | Type
---+---------------------------+-----------------------------------------------------------------------+--------
i+ | nvidia-computeG05         | NVIDIA driver for computing with GPGPU                                | package
i  | nvidia-gfxG05-kmp-default | NVIDIA graphics driver kernel module for GeForce 600 series and newer | package
i  | nvidia-glG05              | NVIDIA OpenGL libraries for OpenGL acceleration                       | package
i  | x11-video-nvidiaG05       | NVIDIA graphics driver for GeForce 600 series and newer               | package

```

Thank you.

```
andy@localhost:~> sudo hwinfo --short
[sudo] password for root: 
cpu:                                                            
                       AMD Ryzen 5 4600H with Radeon Graphics, 1400 MHz
                       AMD Ryzen 5 4600H with Radeon Graphics, 1400 MHz
                       AMD Ryzen 5 4600H with Radeon Graphics, 1400 MHz
                       AMD Ryzen 5 4600H with Radeon Graphics, 1530 MHz
                       AMD Ryzen 5 4600H with Radeon Graphics, 1400 MHz
                       AMD Ryzen 5 4600H with Radeon Graphics, 3000 MHz
                       AMD Ryzen 5 4600H with Radeon Graphics, 1400 MHz
                       AMD Ryzen 5 4600H with Radeon Graphics, 1400 MHz
                       AMD Ryzen 5 4600H with Radeon Graphics, 1400 MHz
                       AMD Ryzen 5 4600H with Radeon Graphics, 1400 MHz
                       AMD Ryzen 5 4600H with Radeon Graphics, 3000 MHz
                       AMD Ryzen 5 4600H with Radeon Graphics, 1400 MHz
keyboard:
  /dev/input/event13   Integrated Technology Express ITE Device(8910)
  /dev/input/event9    Shenzhen Rapoo Rapoo Gaming Mouse
  /dev/input/event8    Integrated Technology Express ITE Device(8295)
  /dev/input/event0    AT Translated Set 2 keyboard
mouse:
  /dev/input/mice      Shenzhen Rapoo Rapoo Gaming Mouse
  /dev/input/mice      MSFT0001:00 06CB:7F28 Mouse
  /dev/input/mice      MSFT0001:00 06CB:7F28 Touchpad
monitor:
                       AUO LCD Monitor
graphics card:
                       nVidia TU117M [GeForce GTX 1650 Ti Mobile]
                       ATI Renoir
sound:
                       AMD Family 17h (Models 10h-1fh) HD Audio Controller
storage:
                       Micron Non-Volatile memory controller
                       AMD FCH SATA Controller [AHCI mode]
                       AMD FCH SATA Controller [AHCI mode]
network:
  eno1                 Realtek RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller
  wlp4s0               Intel Wi-Fi 6 AX200NGW
network interface:
  eno1                 Ethernet network interface
  lo                   Loopback network interface
  wlp4s0               Ethernet network interface
disk:
  /dev/nvme0n1         Micron Disk
partition:
  /dev/nvme0n1p1       Partition
  /dev/nvme0n1p2       Partition
  /dev/nvme0n1p3       Partition
usb controller:
                       AMD Renoir USB 3.1
                       AMD Renoir USB 3.1
bios:
                       BIOS
bridge:
                       AMD Renoir PCIe Dummy Host Bridge
                       AMD Renoir Device 24: Function 3
                       AMD Renoir PCIe GPP Bridge
                       AMD Renoir Device 24: Function 1
                       AMD Renoir PCIe Dummy Host Bridge
                       AMD FCH LPC Bridge
                       AMD Renoir PCIe GPP Bridge
                       AMD Renoir Device 24: Function 6
                       AMD Renoir Root Complex
                       AMD Renoir Internal PCIe GPP Bridge to Bus
                       AMD Renoir Device 24: Function 4
                       AMD Renoir Device 24: Function 2
                       AMD Renoir PCIe GPP Bridge
                       AMD Renoir Device 24: Function 0
                       AMD Renoir PCIe GPP Bridge
                       AMD Renoir PCIe Dummy Host Bridge
                       AMD Renoir Device 24: Function 7
                       AMD Renoir Internal PCIe GPP Bridge to Bus
                       AMD Renoir Device 24: Function 5
hub:
                       Linux Foundation 2.0 root hub
                       Genesys Logic Hub
                       Linux Foundation 3.0 root hub
                       Linux Foundation 2.0 root hub
                       Genesys Logic USB3.2 Hub
                       Linux Foundation 3.0 root hub
memory:
                       Main Memory
bluetooth:
                       Intel AX200 Bluetooth
unknown:
                       FPU
                       DMA controller
                       PIC
                       Keyboard controller
                       PS/2 Controller
                       AMD Renoir IOMMU
                       AMD Family 17h (Models 10h-1fh) Platform Security Processor
                       AMD FCH SMBus Controller
                       AMD Raven/Raven2/FireFlight/Renoir Audio Processor
                       Integrated Technology Express ITE Device(8295)
                       Shenzhen Rapoo Rapoo Gaming Mouse
  /dev/input/event20   Syntek Integrated Camera

```

**Comment by sndirsch** [comment]
Apparently you have both. AMD Renoir (integrated graphics) and nVidia GTX1650 discrete graphics.

> bbswitch module not found. NVIDIA card will not be powered off
This is a bug in the script, I will fix. You can just ignore it.

Have you logged out from your Xsession and logged in again or rebooted your machine after your ran `prime-select nvidia` the first time? If you did I'm afraid the nvidia driver on your machine does not work in general for some reason. I suggest to attach the X logfile and output of `dmesg`.

**Comment by kevinsmia1939** [comment]
> Apparently you have both. AMD Renoir (integrated graphics) and nVidia GTX1650 discrete graphics.
> 
> > bbswitch module not found. NVIDIA card will not be powered off
> 
> This is a bug in the script, I will fix. You can just ignore it.
> 
> Have you logged out from your Xsession and logged in again or rebooted your machine after your ran `prime-select nvidia` the first time? If you did I'm afraid the nvidia driver on your machine does not work in general for some reason. I suggest to attach the X logfile and output of `dmesg`.

Hi, 
I tried restarting and logout after running `prime-select nvidia`, same issue.
dmesg output https://pastebin.com/RPFCde3p
my Xorg.0.log https://pastebin.com/JM2XiyVD

**Comment by sndirsch** [comment]
Hmm. No traces of nVidia driver. As if this driver hasn't been installed. Please provide output of `prime-select log-view`

**Comment by kevinsmia1939** [comment]
> prime-select log-view

##SUSEPrime logfile##
[ 22:52:01 ] user_logout_waiter: started
[ 22:52:14 ] user_logout_waiter: X restart detected, preparing switch to nvidia
/var/log/prime-select.log (END)

**Comment by sndirsch** [comment]
Hmm. This looks ok. I suggest to reinstall the nvidia driver

```
rpm -e nvidia-gfxG05-kmp-default --nodeps
zypper in nvidia-gfxG05-kmp-default
```

Try to load the driver

```
dmesg -c > /dev/null
modprobe nvidia
dmesg
```

Send me the output of last dmesg command.

Then switch back to 'amd" and then again to "nvidia"

```
prime-select amd
<logout>
prime-select get-current
prime-select nvidia
<logout>
prime-select get-current
```

Sorry, I have no better ideas right now ...

**Comment by sndirsch** [comment]
Ok. So let's close as fixed.

> > bbswitch module not found. NVIDIA card will not be powered off
> 
> This is a bug in the script, I will fix. You can just ignore it.

This is fixed now in release 0.8.7.
